src/coreclr/jit/fgprofile.cpp

```diff
@@ -3341,6 +3341,11 @@ void EfficientEdgeCountReconstructor::Solve()
     unsigned       nPasses = 0;
     unsigned const nLimit  = 10;
 
+    // If we end up with negative edge weights because of inconsistent counts, we use this factor to
+    // substitute a small positive one.
+    //
+    weight_t const smallFactor = 0.001;
```

Comment:
there is ProfileSynthesis.epsilon that probably can be used here?

Reply:
Yea, good idea. Updated.